pkg/checks/external/nodeCheck/main.go

```diff
@@ -0,0 +1,211 @@
+package nodeCheck
+
+import (
+	"context"
+	"errors"
+	"net"
+	"os"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/Comcast/kuberhealthy/v2/pkg/checks/external"
+)
+
+// EnableDebugOutput enables debug logging for this package
+func EnableDebugOutput() {
+	log.SetLevel(log.DebugLevel)
+}
+
+// WaitForKuberhealthy waits for the the kuberhealthy endpoint (KH_REPORTING_URL) to be contactable by the checker pod
+// on a given node
+func WaitForKuberhealthy(ctx context.Context) error {
+
+	kuberhealthyEndpoint := os.Getenv(external.KHReportingURL)
+
+	// check the length of the reporting url to make sure we pulled one properly
+	if len(kuberhealthyEndpoint) < 1 {
+		return errors.New("error getting kuberhealthy reporting URL from environment variable " +
+			external.KHReportingURL + " was blank")
+	}
+
+	log.Debugln("Checking if the kuberhealthy endpoint:", kuberhealthyEndpoint, "is ready.")
+	select {
+	case err := <-waitForKuberhealthyEndpointReady(kuberhealthyEndpoint, ctx):
+		if err != nil {
+			return err
+		}
+		log.Debugln("Kuberhealthy endpoint:", kuberhealthyEndpoint, "is ready. Proceeding to run check.")
+	case <-ctx.Done():
+		return errors.New("context cancelled waiting for Kuberhealthy endpoint to be ready")
+	}
+	return nil
+}
+
+// WaitForKubeProxy waits for kube proxy to be ready and running on the node before running the check. Assumes that the
+// kube-proxy pod follows the naming convention: "kube-proxy-{nodeName}"
+func WaitForKubeProxy(client *kubernetes.Clientset, namespace string, ctx context.Context) error {
+
+	khPod, err := getKHPod(client, namespace)
+	if err != nil {
+		return errors.New("error getting kuberhealthy pod: " + err.Error())
+	}
+	log.Debugln("Checking if kube-proxy is running and ready on this node:", khPod.Spec.NodeName)
+	select {
+	case err := <- waitForKubeProxyReady(client, khPod.Spec.NodeName, ctx):
+		if err != nil {
+			return err
+		}
+		log.Debugln("Kube proxy is ready. Proceeding to run check.")
+	case <-ctx.Done():
+		return errors.New("context cancelled waiting for kube proxy to be ready and running")
+	}
+	return nil
+}
+
+// WaitForNodeAge checks the node's age to see if its less than the minimum node age. If so, sleeps for a given sleep duration.
+func WaitForNodeAge(client *kubernetes.Clientset, namespace string, minNodeAge time.Duration, sleepDuration time.Duration, ctx context.Context) error {
```

Comment:
Lets make this check repeatedly ask if the node is new until it's not.  That way, when you call it, it will block until the node is old enough.  Don't worry about sleeping here.  The user can do that in the calling code.

Reply:
instead of asking the node repeatedly if its new or not -- can we just sleep until it reaches the minimum node age?

Reply:
Ah yeah, then re-check.  Good idea.

- fetch node age
- calculate when node should be old enough
- sleep the remaining duration
- re-check node age after sleeping? (maybe not)

Reply:
imo idt we need to re-check? unless we want to be SURE its past its minNodeAge...?